### PR TITLE
[receiver/kafka] unexport the function `WithTracesUnmarshalers`, `WithMetricsUnmarshalers`, `WithLogsUnmarshalers`

### PR DESCRIPTION
**Pull request by JaredTan95**
**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26304

### CODE REVIEW

**Review by songy23 — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
This looks ok to me, pinging codeowners @pavolloffay @MovieStoreGuy

**Comment by codeboten** [comment]
One more ping codeowners @pavolloffay @MovieStoreGuy

**Comment by JaredTan95** [comment]
> Hello @JaredTan95 , it seems need to resolve the conflicts?

done

File: cmd/checkapi/allowlist.txt [hunk 1]
```diff
@@ -7,4 +7,4 @@ receiver/collectdreceiver
 receiver/dockerstatsreceiver
 receiver/journaldreceiver
 receiver/kafkareceiver
```
Comment (sakulali) [inline cmd/checkapi/allowlist.txt:9]: Should we remove kafkareceiver in allowlist.txt?  Then we can run command `go run cmd/checkapi/main.go . ` to confirm again

Comment (sakulali) [inline cmd/checkapi/allowlist.txt:9]:
```suggestion
```
Should we remove kafkareceiver in allowlist.txt? Then we can run command go run cmd/checkapi/main.go .  to confirm again

**Comment by sakulali** [comment]
After resolve the conflicts, i think this PR is ready for merge?

**Comment by JaredTan95** [comment]
> After resolve the conflicts, i think this PR is ready for merge?

resolved

**Review by codeboten — APPROVED** [review]

**Pull request merged**